openedx/core/djangoapps/content_tagging/rules.py

```diff
@@ -262,6 +263,50 @@ def can_view_object_tag_objectid(user: UserType, object_id: str) -> bool:
     return has_studio_read_access(user, course_key)
 
 
+@rules.predicate
+def can_change_object_tag(
+    user: UserType, perm_obj: oel_tagging.ObjectTagPermissionItem | None = None
+) -> bool:
+    """
+    Checks if the user has permissions to create or modify tags on the given taxonomy and object_id.
+    """
+    if not oel_tagging.can_change_object_tag(user, perm_obj):
+        return False
+
+    # The following code allows METHOD permission (PUT) in the viewset for everyone
+    if perm_obj is None:
+        return True
+
+    # TaxonomySerializer use this rule passing object_id = "" to check if the user
+    # can use the taxonomy
+    if perm_obj.object_id == "":
+        return True
+
+    # Also skip taxonomy check if the taxonomy is not set
+    if not perm_obj.taxonomy:
+        return True
+
+    # Taxonomy admins can tag any object using any taxonomy
+    if oel_tagging.is_taxonomy_admin(user):
+        return True
```

Comment:
We can avoid the "Taxonomy admins can tag any object using any taxonomy" issue like this, however: this change bumps the [query count for updating object tags](https://github.com/openedx/edx-platform/blob/a9bd61d01546a6775cd2ecfbdf1e49ad5c8e866e/openedx/core/djangoapps/content_tagging/rest_api/v1/tests/test_views.py#L1618) up from 7 to 13!

```suggestion
    # Only Taxonomy admins can tag objects using a no-org taxonomy
    if not perm_obj.taxonomy.taxonomyorg_set.exists():
        return oel_tagging.is_taxonomy_admin(user)
```

So I can't actually recommend this change. But, I have a task next sprint to fix these query counts, so I'll handle making this change if needed.

Reply:
Besides the query count, I wonder if it is worth making this differentiation: if adminA adds this taxonomy to an org, does this change the permission from adminB?

I'm uncomfortable with this, but I think treating no-org and cross-org the same way (allowing or blocking) would be the best path here.

I commented on the issue you created about the query count.
The staff account is probably the best-case scenario. But I'm sure that you will find what we got wrong.